tests/components/mqtt_statestream/test_init.py

```diff
@@ -3,375 +3,354 @@
+        call.async_publish("pub/another/entity/last_changed", ANY, 1, True),
+        call.async_publish("pub/another/entity/last_updated", ANY, 1, True),
+    ]
+
+    mqtt_mock.async_publish.assert_has_calls(calls, any_order=True)
+    assert mqtt_mock.async_publish.called
+
+
+async def test_state_changed_attr_sends_message(hass, mqtt_mock):
+    """Test the sending of a new message if attribute changed."""
+    e_id = "fake.entity"
+    base_topic = "pub"
+
+    # Add the statestream component for publishing state updates
+    assert await add_statestream(hass, base_topic=base_topic, publish_attributes=True)
+    await hass.async_block_till_done()
+
+    # Reset the mock because it will have already gotten calls for the
+    # mqtt_statestream state change on initialization, etc.
+    mqtt_mock.async_publish.reset_mock()
+
+    test_attributes = {"testing": "YES", "list": ["a", "b", "c"], "bool": False}
+
+    # Set a state of an entity
+    mock_state_change_event(hass, State(e_id, "off", attributes=test_attributes))
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+
+    # Make sure 'on' was published to pub/fake/entity/state
+    calls = [
+        call.async_publish("pub/fake/entity/state", "off", 1, True),
+        call.async_publish("pub/fake/entity/testing", '"YES"', 1, True),
+        call.async_publish("pub/fake/entity/list", '["a", "b", "c"]', 1, True),
+        call.async_publish("pub/fake/entity/bool", "false", 1, True),
+    ]
+
+    mqtt_mock.async_publish.assert_has_calls(calls, any_order=True)
+    assert mqtt_mock.async_publish.called
+
+
+async def test_state_changed_event_include_domain(hass, mqtt_mock):
+    """Test that filtering on included domain works as expected."""
+    base_topic = "pub"
+
+    incl = {"domains": ["fake"]}
+    excl = {}
+
+    # Add the statestream component for publishing state updates
+    # Set the filter to allow fake.* items
+    assert await add_statestream(
+        hass, base_topic=base_topic, publish_include=incl, publish_exclude=excl
+    )
+    await hass.async_block_till_done()
+
+    # Reset the mock because it will have already gotten calls for the
+    # mqtt_statestream state change on initialization, etc.
+    mqtt_mock.async_publish.reset_mock()
+
+    # Set a state of an entity
+    mock_state_change_event(hass, State("fake.entity", "on"))
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+
+    # Make sure 'on' was published to pub/fake/entity/state
+    mqtt_mock.async_publish.assert_called_with("pub/fake/entity/state", "on", 1, True)
+    assert mqtt_mock.async_publish.called
+
+    mqtt_mock.async_publish.reset_mock()
+    # Set a state of an entity that shouldn't be included
+    mock_state_change_event(hass, State("fake2.entity", "on"))
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+
+    assert not mqtt_mock.async_publish.called
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)